### PR TITLE
Add Application EGO with qEI approx for parallel computing

### PR DESCRIPTION
**Pull request by EmileRouxSMB**
The application _EGOp_ implement the _q-EI_ criterion well adapted for parallel evaluation of the function to minimize.
The _EGOp_ Class inherit from the _EGO_ one.
The class embeds a basic parallel job manager based on `joblib`.
A tutorial comes with the application, if necessary I can make more documentation or example.

Best

<p> Tipical use : </p>

```python
criterion='EI' #'EI' or 'SBO' or 'UCB'

#number of points in the initial DOE
ndoe = 9 #(at least ndim+1)

#number of iterations with EGO 
n_iter = 15

#number of sub-iterations with EGO-p (number of samples extracting with the q-EI criterion and the Kriging believer method)
n_par = 2

#Build the initial DOE
sampling = LHS(xlimits=xlimits)
xdoe = sampling(ndoe)


#EGO call
ego = EGO_para(n_iter=n_iter,
               criterion=criterion,
               xdoe=xdoe,
               xlimits=xlimits,
               verbose=True,
               n_par=n_par)

x_opt, y_opt, ind_best, x_data, y_data, x_doe, y_doe, krig = ego.optimize(fun=test_func)
```

### CODE REVIEW

**Comment by EmileRouxSMB** [comment]
I just figure out that an other pull request implement the same algorithm.
It may be better to close this one. I will be glade to help later on. 
Never the less it seemd that their is a complex erreur in the travis os-py27 test. 
Regards

**Comment by relf** [comment]
Thanks for your contribution, with the PR #190 and yours, EGO with qEI should make its way in SMT applications.

**Comment by EmileRouxSMB** [comment]
Add doc for the application 

egop_TestEGOp_run_egop_example

**Comment by relf** [comment]
Hi Emile. I am going to integrate your EGOp in the EGO class as there is no point in having two different classes. I am going to pull your changes and open a new pull request.

File: smt/surrogate_models/idwclib.py [hunk 1]
```diff
@@ -0,0 +1,7 @@
+def __bootstrap__():
```
Comment (relf) [inline smt/surrogate_models/idwclib.py:1]: Not related to the current topic. Why did you add this?

File: smt/applications/egop.py [hunk 2]
```diff
@@ -0,0 +1,240 @@
...
+        declare("n_iter", None, types=int, desc="Number of optimizer steps")
+        declare(
+            "n_max_optim",
+            20,
+            types=int,
+            desc="Maximum number of internal optimizations",
+        )
+        declare("n_start", 20, types=int, desc="Number of optimization start points")
+        declare("n_par", 1, types=int, desc="Number parallel sample the compute using the qEI ")
+        declare("qEIAproxCrit", "KBLB", types=str,
```
Comment (relf) [inline smt/applications/egop.py:58]: In an attempt  to have a consistent naming, I wil rename the option <code>q_EI</code>

File: smt/applications/egop.py [hunk 3]
```diff
@@ -0,0 +1,240 @@
...
+            None,
+            types=int,
+            desc="Number of points of the initial LHS doe, only used if xdoe is not given",
+        )
+        declare("xdoe", None, types=np.ndarray, desc="Initial doe inputs")
+        declare("ydoe", None, types=np.ndarray, desc="Initial doe outputs")
+        declare("xlimits", None, types=np.ndarray, desc="Bounds of function fun inputs")
+        declare("verbose", False, types=bool, desc="Print computation information")
+
+    def optimize(self, fun, JobRunner=None, **kwargs):
```
Comment (relf) [inline smt/applications/egop.py:72]: I noticed you have suppressed JobRunner class as it was not py27 compatible but we may re-introduce it as we are heading to a drop of py2.7 support and state that this feature works only in Python 3.

File: doc/_src_docs/applications/egop.rstx [hunk 4]
```diff
@@ -0,0 +1,73 @@
+Efficient Global Optimization parallel version (EGOp)
```
Comment (relf) [inline doc/_src_docs/applications/egop.rstx:1]: I will integrate this with current EGO doc.

**Comment by relf** [comment]
Thanks @EmileRouxSMB . This PR is integrated in PR #202
